examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2926 @@
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE.md at
+ * the top level directory of deal.II.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Authors: Matthias Maier, Texas A&M University;
+ *          Ignacio Tomas, Texas A&M University, Sandia National Laboratories
+ *
+ * Sandia National Laboratories is a multimission laboratory managed and
+ * operated by National Technology & Engineering Solutions of Sandia, LLC, a
+ * wholly owned subsidiary of Honeywell International Inc., for the U.S.
+ * Department of Energy's National Nuclear Security Administration under
+ * contract DE-NA0003525. This document describes objective technical results
+ * and analysis. Any subjective views or opinions that might be expressed in
+ * the paper do not necessarily represent the views of the U.S. Department of
+ * Energy or the United States Government.
+ */
+
+// @sect3{Include files}
+
+// The set of include files is quite standard. The most intriguing part is
+// the fact that we will rely solely on deal.II data structures for MPI
+// parallelization, in particular distributed::Triangulation and
+// LinearAlgebra::distributed::Vector included through
+// <code>distributed/tria.h</code> and
+// <code>lac/la_parallel_vector.h</code>. Instead of a Trilinos, or PETSc
+// specific matrix class, we will use a non-distributed
+// dealii::SparseMatrix (<code>lac/sparse_matrix.h</code>) to store the local
+// part of the $\mathbf{c}_{ij}$, $\mathbf{n}_{ij}$ and $d_{ij}$ matrices.
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/parallel.h>
+#include <deal.II/base/parameter_acceptor.h>
+#include <deal.II/base/partitioner.h>
+#include <deal.II/base/quadrature.h>
+#include <deal.II/base/timer.h>
+#include <deal.II/base/work_stream.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping.h>
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/manifold_lib.h>
+
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/la_parallel_vector.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/sparse_matrix.templates.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/meshworker/scratch_data.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+
+// In addition to above deal.II specific includes, we also include four
+// boost headers. The first two are for binary archives that we will use
+// for implementing a check-pointing and restart mechanism.
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
```

Comment:
do you think you could add a short section to the introduction that outlines the big ideas of how to achieve checkpoint/restart (and why one would want to use it)?

Reply:
I will add such a discussion sometime later this week.

Reply:
Did you get around to this? I can't seem to find it.